### PR TITLE
en, zh: improve security documents

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added, or deleted? (Required)

- Improve documents related to security
- Update outdated TiCDC links

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/containers-run-as-non-root-user.md [hunk 1]
```diff
@@ -23,34 +23,32 @@ controllerManager:
...
-      fsGroup: 2000
-  tidb:
-    podSecurityContext:
-      runAsUser: 1000
-      runAsGroup: 2000
-      fsGroup: 2000
-```
-
-如果同时配置了集群级别和组件级别，则该组件以组件级别的配置为准。
+对于按照 CR 生成的容器，你同样可以在任意一种 CR (`TidbCluster`/`DMCluster`/`TiInitializer`/`TiMonitor`/`Backup`/`BackupSchedule`/`Restore`) 中配置安全上下文 (security context)。
```
Comment (ran-huang) [inline zh/containers-run-as-non-root-user.md:26]: 其他文档里只见过 `TidbInitializer`/`TidbMonitor`，这里原文是正确的吗？ @DanielZhangQD

File: zh/containers-run-as-non-root-user.md [hunk 2]
```diff
@@ -23,34 +23,32 @@ controllerManager:
...
+
+    ```yaml
+    spec:
+      podSecurityContext:
+        runAsUser: 1000
+        runAsGroup: 2000
+        fsGroup: 2000
+    ```
+
+- 配置在组件级别，仅对该组件生效。 例如，配置 `TidbCluster` 的 `spec.tidb.podSecurityContext`，配置 `DMCluster` 的 `spec.master.podSecurityContext`)。配置示例如下：
```
Comment (ran-huang) [inline zh/containers-run-as-non-root-user.md:40]:
```suggestion
- 配置在组件级别，仅对该组件生效。例如，为 PD 组件配置 `spec.pd.podSecurityContext`，为 TiDB 组件配置 `spec.tidb.podSecurityContext`。配置示例如下：
```
这里文字和 YAML 的示例不一致，可能引起用户困惑？

File: zh/containers-run-as-non-root-user.md (hunk 1 above)
Comment (csuzhangxc) [reply zh/containers-run-as-non-root-user.md:26]: 应该是 `TidbInitializer`, `TidbMonitor` 以及 `DmCluster`

**Review by csuzhangxc — APPROVED** [review]

**Comment by lilin90** [comment]
@ran-huang Comments resolved. PTAL again~ Thanks!

**Review by ran-huang — APPROVED** [review]
LGTM

File: en/containers-run-as-non-root-user.md [hunk 3]
```diff
@@ -23,34 +23,32 @@ controllerManager:
...
-      fsGroup: 2000
-  tidb:
-    podSecurityContext:
-      runAsUser: 1000
-      runAsGroup: 2000
-      fsGroup: 2000
-```
-
-For a component, if both the cluster level and the component level are configured, only the configuration of the component level takes effect.
+For the containers controlled by CR, you can configure security context in any CRs (`TidbCluster`/`DmCluster`/`TidbInitializer`/`TidbMonitor`/`Backup`/`BackupSchedule`/`Restore`) to make the containers run as a non-root user.
```
Comment (lilin90) [inline en/containers-run-as-non-root-user.md:26]:
```suggestion
For the containers controlled by Custom Resource (CR), you can configure security context in any CRs (`TidbCluster`/`DmCluster`/`TidbInitializer`/`TidbMonitor`/`Backup`/`BackupSchedule`/`Restore`) to make the containers run as a non-root user.
```

File: zh/containers-run-as-non-root-user.md [hunk 4]
```diff
@@ -23,34 +23,32 @@ controllerManager:
...
-      fsGroup: 2000
-  tidb:
-    podSecurityContext:
-      runAsUser: 1000
-      runAsGroup: 2000
-      fsGroup: 2000
-```
-
-如果同时配置了集群级别和组件级别，则该组件以组件级别的配置为准。
+对于按照 CR 生成的容器，你同样可以在任意一种 CR (`TidbCluster`/`DmCluster`/`TidbInitializer`/`TidbMonitor`/`Backup`/`BackupSchedule`/`Restore`) 中配置安全上下文 (security context)。
```
Comment (lilin90) [inline zh/containers-run-as-non-root-user.md:26]:
```suggestion
对于按照 CR 生成的容器，你同样可以在任意一种 Custom Resource (CR) (`TidbCluster`/`DmCluster`/`TidbInitializer`/`TidbMonitor`/`Backup`/`BackupSchedule`/`Restore`) 中配置安全上下文 (security context)。
```

**Pull request merged**